diff -Nur qtbase-opensource-src-5.5.1/src/network/socket/qnativesocketengine_unix.cpp qtbase-opensource-src-5.5.1n/src/network/socket/qnativesocketengine_unix.cpp
--- qtbase-opensource-src-5.5.1/src/network/socket/qnativesocketengine_unix.cpp	2015-10-13 12:35:28.000000000 +0800
+++ qtbase-opensource-src-5.5.1n/src/network/socket/qnativesocketengine_unix.cpp	2015-11-07 01:42:18.607963823 +0800
@@ -337,7 +337,7 @@
 
     int n, level;
     convertToLevelAndOption(opt, socketProtocol, level, n);
-#if defined(SO_REUSEPORT)
+#if defined(SO_REUSEPORT) && !defined(Q_OS_LINUX)
     if (opt == QNativeSocketEngine::AddressReusable) {
         // on OS X, SO_REUSEADDR isn't sufficient to allow multiple binds to the
         // same port (which is useful for multicast UDP). SO_REUSEPORT is, but
