--- qtbase-opensource-src-5.5.1/src/plugins/platforms/xcb/qxcbconnection.cpp	2015-10-13 12:35:27.000000000 +0800
+++ qtbase-opensource-src-5.5.1n/src/plugins/platforms/xcb/qxcbconnection.cpp	2015-10-19 17:23:23.965134440 +0800
@@ -1442,7 +1442,6 @@
     int connection_error = xcb_connection_has_error(xcb_connection());
     if (connection_error) {
         qWarning("The X11 connection broke (error %d). Did the X11 server die?", connection_error);
-        exit(1);
     }
 
     QXcbEventArray *eventqueue = m_reader->lock();
