diff --git a/src/gui/kernel/qwindow.cpp b/src/gui/kernel/qwindow.cpp
index 68bcbde..42302db 100644
--- a/src/gui/kernel/qwindow.cpp
+++ b/src/gui/kernel/qwindow.cpp
@@ -1159,10 +1159,10 @@ Qt::WindowState QWindow::windowState() const
 void QWindow::setTransientParent(QWindow *parent)
 {
     Q_D(QWindow);
-    if (parent && !parent->isTopLevel()) {
+    /*if (parent && !parent->isTopLevel()) {
         qWarning() << parent << "must be a top level window.";
         return;
-    }
+    }*/
 
     d->transientParent = parent;
 
diff --git a/src/widgets/kernel/qwidget.cpp b/src/widgets/kernel/qwidget.cpp
index c57ca41..b2992c6 100644
--- a/src/widgets/kernel/qwidget.cpp
+++ b/src/widgets/kernel/qwidget.cpp
@@ -1453,7 +1453,8 @@ void QWidgetPrivate::create_sys(WId window, bool initializeWindow, bool destroyO
     if (QWidget *nativeParent = q->nativeParentWidget()) {
         if (nativeParent->windowHandle()) {
             if (flags & Qt::Window) {
-                win->setTransientParent(nativeParent->window()->windowHandle());
+                //win->setTransientParent(nativeParent->window()->windowHandle());
+                win->setTransientParent(nativeParent->windowHandle());
                 win->setParent(0);
             } else {
                 win->setTransientParent(0);
