diff -Nur crypto-policies/Makefile crypto-policiesn/Makefile
--- crypto-policies/Makefile	2015-10-05 07:57:26.000000000 +0000
+++ crypto-policiesn/Makefile	2015-10-25 08:22:50.961273371 +0000
@@ -17,7 +17,7 @@
 
 update-crypto-policies.8: update-crypto-policies.8.txt
 	asciidoc.py -v -d manpage -b docbook update-crypto-policies.8.txt
-	xsltproc --nonet -o update-crypto-policies.8 /usr/share/asciidoc/docbook-xsl/manpage.xsl update-crypto-policies.8.xml
+	xsltproc --nonet -o update-crypto-policies.8 /etc/asciidoc/docbook-xsl/manpage.xsl update-crypto-policies.8.xml
 
 dist:
 	rm -rf crypto-policies && git clone . crypto-policies && rm -rf crypto-policies/.git/ && tar -czf crypto-policies-git${VERSION}.tar.gz crypto-policies
