diff --git a/docs/sphinx/conf.py b/docs/sphinx/conf.py
index 275fbcc..59a3214 100644
--- a/docs/sphinx/conf.py
+++ b/docs/sphinx/conf.py
@@ -396,7 +396,14 @@ pygments_style = 'sphinx'
 
 # The theme to use for HTML and HTML Help pages.  See the documentation for
 # a list of builtin themes.
-html_theme = 'classic'
+try:
+	from sphinx import version_info
+except ImportError:
+	version_info = None
+if version_info and (1, 3) <= version_info:
+	html_theme = 'classic'
+else:
+	html_theme = 'default'
 
 # Theme options are theme-specific and customize the look and feel of a theme
 # further.  For a list of options available for each theme, see the
