From 5e6708be79ff90cf483bfc66e30437d5019404a0 Mon Sep 17 00:00:00 2001
From: Peng Wu <alexepico@gmail.com>
Date: Mon, 11 Nov 2013 15:38:35 +0800
Subject: [PATCH] fixes pinyin properties

---
 src/PYPinyinProperties.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/PYPinyinProperties.cc b/src/PYPinyinProperties.cc
index 66205de..2d562cc 100644
--- a/src/PYPinyinProperties.cc
+++ b/src/PYPinyinProperties.cc
@@ -189,7 +189,7 @@ PinyinProperties::reset (void)
 
 gboolean
 PinyinProperties::propertyActivate (const gchar *prop_name, guint prop_state) {
-    const static std::string mode_chinese ("mode.chinese");
+    const static std::string mode_chinese ("InputMode");
     const static std::string mode_full ("mode.full");
     const static std::string mode_full_punct ("mode.full_punct");
     const static std::string mode_simp ("mode.simp");
-- 
1.8.3.1

