diff -Nur cryptography-0.9/src/cryptography/hazmat/backends/openssl/dsa.py cryptography-0.9.1/src/cryptography/hazmat/backends/openssl/dsa.py
--- cryptography-0.9/src/cryptography/hazmat/backends/openssl/dsa.py	2015-05-14 09:29:06.000000000 +0800
+++ cryptography-0.9.1/src/cryptography/hazmat/backends/openssl/dsa.py	2015-06-07 04:29:18.000000000 +0800
@@ -40,13 +40,10 @@
         self._hash_ctx.update(data)
 
     def verify(self):
-        self._dsa_cdata = self._backend._ffi.gc(self._public_key._dsa_cdata,
-                                                self._backend._lib.DSA_free)
-
         data_to_verify = self._hash_ctx.finalize()
 
         data_to_verify = _truncate_digest_for_dsa(
-            self._dsa_cdata, data_to_verify, self._backend
+            self._public_key._dsa_cdata, data_to_verify, self._backend
         )
 
         # The first parameter passed to DSA_verify is unused by OpenSSL but
diff -Nur cryptography-0.9/src/cryptography/hazmat/bindings/openssl/x509.py cryptography-0.9.1/src/cryptography/hazmat/bindings/openssl/x509.py
--- cryptography-0.9/src/cryptography/hazmat/bindings/openssl/x509.py	2015-05-14 09:29:06.000000000 +0800
+++ cryptography-0.9.1/src/cryptography/hazmat/bindings/openssl/x509.py	2015-06-07 04:29:24.000000000 +0800
@@ -292,8 +292,8 @@
 /* These aren't macros these arguments are all const X on openssl > 1.0.x */
 int X509_CRL_set_lastUpdate(X509_CRL *, ASN1_TIME *);
 int X509_CRL_set_nextUpdate(X509_CRL *, ASN1_TIME *);
-int X509_set_notBefore(X509 *, ASN1_UTCTIME *);
-int X509_set_notAfter(X509 *, ASN1_UTCTIME *);
+int X509_set_notBefore(X509 *, ASN1_TIME *);
+int X509_set_notAfter(X509 *, ASN1_TIME *);
 
 /* These use STACK_OF(X509_EXTENSION) in 0.9.8e. Once we drop support for
    RHEL/CentOS 5 we should move these back to FUNCTIONS. */
