diff --git a/examples/logrotate.cron b/examples/logrotate.cron
index c6d50d4..967932e 100644
--- a/examples/logrotate.cron
+++ b/examples/logrotate.cron
@@ -1,6 +1,6 @@
 #!/bin/sh
 
-/usr/sbin/logrotate /etc/logrotate.conf
+/usr/sbin/logrotate -s /var/lib/logrotate/logrotate.status /etc/logrotate.conf
 EXITVALUE=$?
 if [ $EXITVALUE != 0 ]; then
     /usr/bin/logger -t logrotate "ALERT exited abnormally with [$EXITVALUE]"
