--- intel-gpu-tools-1.11/man/Makefile.am	2015-06-05 08:46:18.000000000 +0000
+++ intel-gpu-tools-1.11n/man/Makefile.am	2015-07-12 10:48:22.957245549 +0000
@@ -21,14 +21,6 @@
 
 appman_DATA = $(appman_PRE:man=$(APP_MAN_SUFFIX))
 
-# man pages in rst, only used if we have rst2man
-appman_RST =				\
-	intel_reg.rst
-
-if HAVE_RST2MAN
-appman_DATA += $(appman_RST:rst=$(APP_MAN_SUFFIX))
-endif
-
 EXTRA_DIST = $(appman_PRE)
 CLEANFILES = $(appman_DATA)
 
