diff -urN m17n-db-1.7.0/MIM/gu-itrans.mim m17n-db-1.7.0-new/MIM/gu-itrans.mim
--- m17n-db-1.7.0/MIM/gu-itrans.mim	2015-05-11 11:33:32.160135550 +0530
+++ m17n-db-1.7.0-new/MIM/gu-itrans.mim	2015-05-08 10:04:48.570562286 +0530
@@ -166,6 +166,7 @@
   ("RRI" (delete @-) "ૄ")		; not in ITRANS Gujarati table
   ("R^I" (delete @-) "ૄ")		; not in ITRANS Gujarati table
   (".c" (delete @-) "ૅ")
+  ("e." (delete @-) "ે.")	        ; not in ITRANS Gujarati table
   ("e.c" (delete @-) "ૅ")		; not in ITRANS Gujarati table
   ("e" (delete @-) "ે")
   ("ai" (delete @-) "ૈ")
diff -urN m17n-db-1.7.0/MIM/mr-itrans.mim m17n-db-1.7.0-new/MIM/mr-itrans.mim
--- m17n-db-1.7.0/MIM/mr-itrans.mim	2014-11-29 04:27:30.000000000 +0530
+++ m17n-db-1.7.0-new/MIM/mr-itrans.mim	2015-05-08 09:22:09.467156393 +0530
@@ -173,6 +173,7 @@
   ("LLi" (delete @-) "ॢ")	      ; not in ITRANS Devanagari table
   ("L^i" (delete @-) "ॢ")	      ; not in ITRANS Devanagari table
   (".c" (delete @-) "ॅ")	      ; not in ITRANS Devanagari table
+  ("e." (delete @-) "े.")	      ; not in ITRANS Devenagari table
   ("e.c" (delete @-) "ॅ")
   ("ee" (delete @-) "ॆ")	      ; not in ITRANS Devanagari table
   ("E" (delete @-) "ॆ")		      ; not in ITRANS Devanagari table
