diff -Nur qt-x11-free-3.3.6/src/kernel/qfontdatabase_x11.cpp qt-x11-free-3.3.6fix/src/kernel/qfontdatabase_x11.cpp
--- qt-x11-free-3.3.6/src/kernel/qfontdatabase_x11.cpp	2006-07-10 21:24:29.000000000 +0800
+++ qt-x11-free-3.3.6fix/src/kernel/qfontdatabase_x11.cpp	2006-07-10 21:24:49.000000000 +0800
@@ -910,9 +910,26 @@
 
 		QtFontStyle::Key key = style->key;
 
+                // does this style have a bold equivalent?
+                key.weight = QFont::Bold;
+                QtFontStyle *equiv = foundry->style(key);
+                if (!equiv) {
+                    // let's fake one...
+                    equiv = foundry->style(key, true);
+                    equiv->smoothScalable = true;
+
+                    QtFontSize *equiv_size = equiv->pixelSize(SMOOTH_SCALABLE, true);
+                    QtFontEncoding *equiv_enc = equiv_size->encodingID(-1, 0, 0, 0, 0, true);
+
+                    // keep the same pitch
+                    equiv_enc->pitch = enc->pitch;
+                   key.weight = QFont::Normal;
+                }
+
+
 		// does this style have an italic equivalent?
 		key.italic = TRUE;
-		QtFontStyle *equiv = foundry->style( key );
+		equiv = foundry->style( key );
 		if ( equiv ) continue;
 
 		// does this style have an oblique equivalent?
