diff -Nur qt-x11-free-3.3.8/src/kernel/qfontdatabase.cpp qt-x11-free-3.3.8n/src/kernel/qfontdatabase.cpp
--- qt-x11-free-3.3.8/src/kernel/qfontdatabase.cpp	2007-09-06 23:02:54.000000000 -0400
+++ qt-x11-free-3.3.8n/src/kernel/qfontdatabase.cpp	2007-09-06 23:03:22.000000000 -0400
@@ -956,7 +956,7 @@
     char pitch = request.ignorePitch ? '*' : request.fixedPitch ? 'm' : 'p';
 
     parseFontName( request.family, foundry_name, family_name );
-
+/*
 #ifdef Q_WS_X11
     if (script == QFont::Han) {
 	// modify script according to locale
@@ -975,7 +975,7 @@
 	script = defaultHan;
     }
 #endif
-
+*/
     FM_DEBUG( "QFontDatabase::findFont\n"
 	      "  request:\n"
 	      "    family: %s [%s], script: %d (%s)\n"
