--- desktop-file-utils-0.12/src/install.c   18 Apr 2006 22:08:10 -0000      1.18
+++ desktop-file-utils-0.12/src/install.c   2 Feb 2007 19:43:27 -0000
@@ -96,12 +96,10 @@ process_one_file (const char *filename,
   GError *rebuild_error;
   GSList *tmp;
   
-  g_assert (vendor_name);
-
   dirname = g_path_get_dirname (filename);
   basename = g_path_get_basename (filename);
   
-  if (!g_str_has_prefix (basename, vendor_name))
+  if (vendor_name && !g_str_has_prefix (basename, vendor_name))
     {
       char *new_base;
       new_base = g_strconcat (vendor_name, "-", basename, NULL);
@@ -564,12 +562,6 @@ main (int argc, char **argv)
   if (vendor_name == NULL)
     vendor_name = g_strdup (g_getenv ("DESKTOP_FILE_VENDOR"));
   
-  if (vendor_name == NULL)
-    {
-      g_printerr (_("Must specify the vendor namespace for these files with --vendor\n"));
-      return 1;
-    }
-
   if (copy_generic_name_to_name && copy_name_to_generic_name)
     {
       g_printerr (_("Specifying both --copy-name-to-generic-name and --copy-generic-name-to-name at once doesn't make much sense.\n"));


