diff -Nur clucene-core-2.3.3.4/src/test/search/TestIndexSearcher.cpp clucene-core-2.3.3.4n/src/test/search/TestIndexSearcher.cpp
--- clucene-core-2.3.3.4/src/test/search/TestIndexSearcher.cpp	2011-03-17 08:21:07.000000000 +0800
+++ clucene-core-2.3.3.4n/src/test/search/TestIndexSearcher.cpp	2013-12-02 15:45:57.182192429 +0800
@@ -5,6 +5,7 @@
 * the GNU Lesser General Public License, as specified in the COPYING file.
 ------------------------------------------------------------------------------*/
 #include "test.h"
+#include <unistd.h>
 
 DEFINE_MUTEX(searchMutex);
 DEFINE_CONDITION(searchCondition);
