diff -Nur libvpx-v1.3.0/build/make/Makefile libvpx-v1.3.0n/build/make/Makefile
--- libvpx-v1.3.0/build/make/Makefile	2014-01-10 20:12:42.000000000 +0000
+++ libvpx-v1.3.0n/build/make/Makefile	2014-10-29 08:00:32.433527783 +0000
@@ -158,7 +158,7 @@
 	$(qexec)$(AS) $(ASFLAGS) -o $@ $<
 
 .PRECIOUS: %.c.S
-%.c.S: CFLAGS += -DINLINE_ASM
+%.c.S: CFLAGS += -DINLINE_ASM -fno-integrated-as
 $(BUILD_PFX)%.c.S: %.c
 	$(if $(quiet),@echo "    [GEN] $@")
 	$(qexec)$(CC) -S $(CFLAGS) -o $@ $<
